src/sql/workbench/contrib/notebook/browser/calloutDialog/imageCalloutDialog.ts

```diff
@@ -0,0 +1,241 @@
+			enabled: true,
+			checked: false
+		});
+		this._imageLocalRadioButton.name = this._editorImageLocationGroup;
+		this._imageLocalRadioButton.value = this.locationLocal;
+		this._imageRemoteRadioButton.name = this._editorImageLocationGroup;
+		this._imageRemoteRadioButton.value = this.locationRemote;
+
+		DOM.append(locationRow, radioButtonGroup);
+
+		let pathRow = DOM.$('.row');
+		DOM.append(imageContentColumn, pathRow);
+		this._imageUrlLabel = DOM.$('p');
+		if (this._imageLocalRadioButton.checked === true) {
+			this._imageUrlLabel.innerText = this.pathPlaceholder;
+		} else {
+			this._imageUrlLabel.innerText = this.urlPlaceholder;
+		}
+		DOM.append(pathRow, this._imageUrlLabel);
+
+		let inputContainer = DOM.$('.flex-container');
+		this._imageUrlInputBox = new InputBox(
+			inputContainer,
+			this._contextViewService,
+			{
+				placeholder: this.pathPlaceholder,
+				ariaLabel: this.pathInputLabel
+			});
+		let browseButtonContainer = DOM.$('.button-icon');
+		this._imageBrowseButton = DOM.$('a.codicon.masked-icon.browse-local');
+		this._imageBrowseButton.title = this.browseAltText;
+		DOM.append(inputContainer, browseButtonContainer);
+		DOM.append(browseButtonContainer, this._imageBrowseButton);
+
+		this._register(DOM.addDisposableListener(this._imageBrowseButton, DOM.EventType.CLICK, async () => {
+			let selectedUri = await this.handleBrowse();
+			if (selectedUri) {
+				this._imageUrlInputBox.value = selectedUri.fsPath;
+			}
+		}, true));
+
+		this._register(this._imageRemoteRadioButton.onClicked(e => {
+			this._imageBrowseButton.style.display = 'none';
+			this._imageUrlLabel.innerText = this.urlPlaceholder;
+			this._imageUrlInputBox.setPlaceHolder(this.urlPlaceholder);
+		}));
+		this._register(this._imageLocalRadioButton.onClicked(e => {
+			this._imageBrowseButton.style.display = 'block';
+			this._imageUrlLabel.innerText = this.pathPlaceholder;
+			this._imageUrlInputBox.setPlaceHolder(this.pathPlaceholder);
+		}));
+		DOM.append(pathRow, inputContainer);
+
+		let embedRow = DOM.$('.row');
+		DOM.append(imageContentColumn, embedRow);
+		this._imageEmbedLabel = DOM.append(embedRow, DOM.$('.checkbox'));
+		this._imageEmbedCheckbox = new Checkbox(
+			this._imageEmbedLabel,
+			{
+				label: this.embedImageLabel,
+				checked: false,
+				onChange: (viaKeyboard) => { },
+				ariaLabel: this.embedImageLabel
+			});
+		DOM.append(embedRow, this._imageEmbedLabel);
+	}
+
+	private registerListeners(): void {
+		this._register(styler.attachInputBoxStyler(this._imageUrlInputBox, this._themeService));
+		this._register(styler.attachCheckboxStyler(this._imageEmbedCheckbox, this._themeService));
+	}
+
+	public insert(): void {
+		this.hide();
+		this._selectionComplete.resolve({
+			insertMarkup: `<img src="${strings.escape(this._imageUrlInputBox.value)}">`,
+			imagePath: this._imageUrlInputBox.value,
+			embedImage: this._imageEmbedCheckbox.checked
+		});
+		this.dispose();
+	}
+
+	public cancel(): void {
+		this.hide();
```

Comment:
You should probably just have cancel defined in the base class that hides and disposes it - and then in the implementations call `super.cancel()` in addition to resolving the promise.